alacritty/src/macos/proc.rs

```diff
@@ -0,0 +1,117 @@
+use std::ffi::{CStr, CString};
+use std::io;
+use std::mem::{self, MaybeUninit};
+use std::os::raw;
+use std::path::PathBuf;
+
+pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
+    let mut info = MaybeUninit::<sys::proc_vnodepathinfo>::uninit();
+    let info_ptr = info.as_mut_ptr() as *mut raw::c_void;
+    let size = mem::size_of::<sys::proc_vnodepathinfo>() as raw::c_int;
+
+    let c_str = unsafe {
+        let pidinfo_size = sys::proc_pidinfo(pid, sys::PROC_PIDVNODEPATHINFO, 0, info_ptr, size);
+        match pidinfo_size {
+            c if c < 0 => return Err(io::Error::last_os_error()),
+            s if s != size => {
+                return Err(io::Error::new(io::ErrorKind::Other, "invalid return size"))
+            },
+            _ => CStr::from_ptr(info.assume_init().pvi_cdir.vip_path.as_ptr()),
+        }
+    };
+
+    CString::from(c_str)
+        .into_string()
+        .map(PathBuf::from)
+        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))
+}
+
+// Cleaned up `bindgen` output below
```

Comment:
What purpose does this comment have? If it actually is useful, which I cannot see at the moment, we should add punctuation at least.

If we actually want to write useful documentation, this should probably be a doc comment explaining that this is a binding to the macos proc system API. Generally internal doc comments are preferred in that case.

Reply:
I'd be happy to expand this comment to include more information about the bindgen process, mainly the command used to create the seed of this file itself. Given that I needed to strip a lot of things manually, I figured this was enough however.

Reply:
I don't think that we should add more docs about a function we're using, we should just refer from where we're using it, since they should have docs already and repeating them doesn't give any benefit.

Reply:
I'm not sure why the bindgen process needs to be documented at all. As mentioned earlier how this code is generated should be irrelevant.

If anything, the module itself should be documented, not what environment/editor/generator you used to create it.